homeassistant/components/broadlink/remote.py

```diff
@@ -0,0 +1,376 @@
+        commands = kwargs[ATTR_COMMAND]
+        device = kwargs[ATTR_DEVICE]
+        repeat = kwargs[ATTR_NUM_REPEATS]
+        delay = kwargs[ATTR_DELAY_SECS]
+
+        if not self._state:
+            return
+
+        should_delay = False
+        for _, cmd in product(range(repeat), commands):
+            try:
+                should_delay = await self._async_send_code(
+                    cmd, device, delay if should_delay else 0
+                )
+            except ConnectionError:
+                break
+
+        self._flag_storage.async_delay_save(self.flags, FLAG_SAVE_DELAY)
+
+    async def _async_send_code(self, command, device, delay):
+        """Send a code to a device.
+
+        For toggle commands, alternate between codes in a list,
+        ensuring that the same code is never sent twice in a row.
+        """
+        try:
+            code = self._codes[device][command]
+        except KeyError:
+            _LOGGER.error("Failed to send '%s/%s': command not found", command, device)
+            return False
+
+        if isinstance(code, list):
+            code = code[self._flags[device]]
+            should_alternate = True
+        else:
+            should_alternate = False
+        await asyncio.sleep(delay)
+
+        try:
+            await self._async_attempt(self._api.send_data, data_packet(code))
+        except ValueError:
+            _LOGGER.error("Failed to send '%s/%s': invalid code", command, device)
+            return False
+        except ConnectionError:
+            _LOGGER.error("Failed to send '%s/%s': remote is offline", command, device)
+            raise
+
+        if should_alternate:
+            self._flags[device] ^= 1
+
+        return True
+
+    async def async_learn_command(self, **kwargs):
+        """Learn a list of commands from a remote."""
+        kwargs = SERVICE_LEARN_SCHEMA(kwargs)
+        commands = kwargs[ATTR_COMMAND]
+        device = kwargs[ATTR_DEVICE]
+        toggle = kwargs[ATTR_ALTERNATIVE]
+        timeout = kwargs[ATTR_TIMEOUT]
+
+        if not self._state:
+            return
+
+        should_store = False
+        for command in commands:
+            try:
+                should_store |= await self._async_learn_code(
+                    command, device, toggle, timeout
+                )
+            except ConnectionError:
+                break
+
+        if should_store:
+            await self._code_storage.async_save(self._codes)
+
+    async def _async_learn_code(self, command, device, toggle, timeout):
+        """Learn a code from a remote.
+
+        Capture an aditional code for toggle commands.
+        """
+        try:
+            code = (
+                await self._async_capture_code(command, timeout)
+                if not toggle
+                else [
+                    await self._async_capture_code(command, timeout),
+                    await self._async_capture_code(command, timeout),
+                ]
+            )
```

Comment:
Please do not use inline if-statements that wrap multiple lines, especially if you are also adding some awaits. Makes it very difficult to understand what's going on here.

Reply:
```python
if toggle:
    code = await …
else:
    code = [
        await …,
        await …,
    ]
```

Reply:
Done.